ts/src/base/Exchange.ts

```diff
@@ -5049,13 +5049,6 @@ export default class Exchange {
         if (!this.substituteCommonCurrencyCodes) {
             return code;
         }
-        // if the provided code already exists as a value in commonCurrencies dict, then we should not again transform it
-        // more details at: https://github.com/ccxt/ccxt/issues/21112#issuecomment-2031293691
-        const commonCurrencies = Object.values (this.commonCurrencies);
-        const exists = this.inArray (code, commonCurrencies);
-        if (exists) {
-            return code;
-        }
```

Comment:
reverted this changed (which was merged few days ago)

Reply:
@ttodua LGTM, but how do we solve issue https://github.com/ccxt/ccxt/issues/21112 ?

Reply:
> @ttodua LGTM, but how do we solve issue [ccxt#21112](https://github.com/ccxt/ccxt/issues/21112) ?

so, removal of `codes = this.marketCodes (codes);` from fDWF method will fix that problem (as that OP was experiencing issue specifically with fDWF method).